webapp/channels/src/components/form_error.tsx

```diff
@@ -11,71 +11,71 @@ type Props = {
     iconClassName?: string;
     margin?: boolean;
     errors?: React.ReactNode[];
-}
+};
 
-export default class FormError extends React.PureComponent<Props> {
-    public static defaultProps = {
-        error: null,
-        errors: [],
-    };
-    public render() {
-        const {error = null, errors = [], iconClassName, margin, textClassName, type} = this.props;
+const EMPTY_ERROR_LIST: React.ReactNode[] = [];
 
-        if (!error && errors.length === 0) {
-            return null;
-        }
+const FormError: React.FC<Props> = ({
```

Comment:
nit: if the functional component is not expecting to handle default children, there is no need to type it as React.FC<Props>, actually I would recomend using `function FormError() {` declaration instead of arrow function declaration since it has benefits specially in the stack trace logging.

Reply:
Just for consistency, we are using all over the place the form `const Component = ({}: Props) {}`, so I would prefer to keep it like this.

Regarding the stack trace though... have you seen this happening and can give some examples? I have also worried about this (specially when in stack traces you see anonymous functions) but I think somewhere in babel / webpack they do magic so these never get anonymous. If we find any good reason to use one way over the other I am happy to change :smile:

Regarding the type definition... yes, it can be removed. In general it should not be needed.

Reply:
Should I change the function back to arrow? @larkox @pvev

Reply:
Let's wait for @pvev answer to my comment :smile: